[Bug] `pip3 install --user wheel "cython<3.0.0"` error on Ubuntu 24.04

### Apache Cloudberry version

main

### What happened

When running `make install -C ~/cloudberry`, errors returned:

```
# Download PyYAML using curl (only if not exists)
Downloading PyYAML-5.4.1.tar.gz...
# install dependencies of gpdemo
for file in gpdemo-defaults.sh lalshell generate_certs.sh demo_cluster.sh probe_config.sh README; do \
	/usr/bin/install -c -m 755 ../../gpAux/gpdemo/$file '/usr/local/cloudberry-db/bin/lib/gpdemo/'$file ; \
done
# Symlink gpcheckcat from bin to bin/lib to maintain backward compatibility
if [ ! -L /usr/local/cloudberry-db/bin/lib/gpcheckcat  ]; then \
	cd /usr/local/cloudberry-db/bin/lib/ && ln -s ../gpcheckcat gpcheckcat; \
fi
/usr/bin/install -c -m 644 gpload.bat '/usr/local/cloudberry-db/bin/gpload.bat'
# Download PyGreSQL using curl (only if not exists)
Downloading PyGreSQL-5.2.tar.gz...
# Install wheel and cython for PyYAML building
pip3 install --user wheel "cython<3.0.0"
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.

    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.

    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.

    See /usr/share/doc/python3.12/README.venv for more information.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
make[2]: *** [Makefile:100: download-python-deps] Error 1
make[2]: Leaving directory '/home/gpadmin/cloudberry/gpMgmt/bin'
make[1]: *** [Makefile:7: install-bin-recurse] Error 2
make[1]: Leaving directory '/home/gpadmin/cloudberry/gpMgmt'
make: *** [GNUmakefile:80: install] Error 2
make: Leaving directory '/home/gpadmin/cloudberry'
```

### What you think should happen instead

_No response_

### How to reproduce

Following the build docs: https://cloudberry.apache.org/docs/next/deployment/quick-build

### Operating System

Ubuntu 24.04.3 LTS

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/apache/cloudberry/blob/main/CODE_OF_CONDUCT.md).

--- a/gpMgmt/bin/Makefile
+++ b/gpMgmt/bin/Makefile
@@ -111,8 +111,19 @@ download-python-deps:
 	else \
 		echo "PyGreSQL-$(PYGRESQL_VERSION).tar.gz already exists, skipping download"; \
 	fi
-	# Install wheel and cython for PyYAML building
-	pip3 install --user wheel "cython<3.0.0"
+	# Install wheel and cython for PyYAML building (only if not exists)
+	@if python3 -c "import wheel" >/dev/null 2>&1; then \
+		echo "wheel already exists, skipping installation"; \
+	else \
+		echo "Installing wheel..."; \
+		pip3 install --user wheel 2>/dev/null || pip3 install --user --break-system-packages wheel; \
+	fi
+	@if python3 -c "import cython" >/dev/null 2>&1; then \
+		echo "cython already exists, skipping installation"; \
+	else \
+		echo "Installing cython..."; \
+		pip3 install --user "cython<3.0.0" 2>/dev/null || pip3 install --user --break-system-packages "cython<3.0.0"; \
+	fi
 
 #
 # PyGreSQL

